price-feeder/oracle/util.go

```diff
@@ -2,48 +2,115 @@ package oracle
 
 import (
 	"fmt"
+	"sort"
+	"time"
 
 	sdk "github.com/cosmos/cosmos-sdk/types"
 	"github.com/umee-network/umee/price-feeder/oracle/provider"
 )
 
+var minimumTimeWeight = sdk.MustNewDecFromStr("0.2")
+
+// tvwapCandlePeriod represents the time period we use for tvwap in minutes.
+const tvwapCandlePeriod = 3 * time.Minute
+
+// compute VWAP for each base by dividing the Σ {P * V} by Σ {V}
+func vwap(weightedPrices map[string]sdk.Dec, volumeSum map[string]sdk.Dec) (map[string]sdk.Dec, error) {
+	for base, p := range weightedPrices {
+		if volumeSum[base].Equal(sdk.ZeroDec()) {
+			return nil, fmt.Errorf("unable to divide by zero")
```

Comment:
Does this mean that is the volumeSum for _any_ denom is zero, that _all_ denoms go dark?

Assuming `base` is a token denomination.

Reply:
Looks like it. I'm not sure what we should do. Should we just filter out the symbol that has zero volume or just error out completely as we do now?

Reply:
We could change the function to something like 

```golang
// compute VWAP for each base by dividing the Σ {P * V} by Σ {V}
func vwap(weightedPrices map[string]sdk.Dec, volumeSum map[string]sdk.Dec) (map[string]sdk.Dec, error) {
	vwap := make(map[string]sdk.Dec)
	for base, p := range weightedPrices {
		if !volumeSum[base].Equal(sdk.ZeroDec()) {
			if _, ok := vwap[base]; !ok {
				vwap[base] = sdk.ZeroDec()
			}

			vwap[base] = p.Quo(volumeSum[base])
		}
	}

	return vwap, nil
}
```

The oracle would then log any missing denoms, wdyt @alexanderbez @toteki ?

Reply:
Done at 1ed0e5e848e539ae1e951412f3cfe7ac29014b21